### PR TITLE
Unify to camel cased setting key

### LINKED ISSUE #3146: Upgrade the keys in Keys to the macros in Def
This would reduce verbosity and enforce the code-name/runtime-name invariant.

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #3146

### CODE REVIEW

File: main/src/main/scala/sbt/Keys.scala [hunk 1]
```diff
@@ -90,176 +90,176 @@ object Keys {
...
-  val initialize = SettingKey[Unit]("initialize", "A convenience setting for performing side-effects during initialization.", BSetting)
-  val onLoad = SettingKey[State => State]("on-load", "Transformation to apply to the build state when the build is loaded.", DSetting)
-  val onUnload = SettingKey[State => State]("on-unload", "Transformation to apply to the build state when the build is unloaded.", DSetting)
-  val onLoadMessage = SettingKey[String]("on-load-message", "Message to display when the project is loaded.", DSetting)
-  val transformState = AttributeKey[State => State]("transform-state", "State transformation to apply after tasks run.", DSetting)
-
-  val onComplete = SettingKey[() => Unit]("on-complete", "Hook to run when task evaluation completes.  The type of this setting is subject to change, pending the resolution of SI-2915.", DSetting)
+  val autoGeneratedProject = settingKey[Boolean]("If it exists, represents that the project (and name) were automatically created, rather than user specified.").withRank(DSetting)
+  val projectCommand = AttributeKey[Boolean]("Marks Commands that were registered for the current Project.", Invisible)
```
Comment (dwijnand) [inline main/src/main/scala/sbt/Keys.scala]: You're making "Marks Commands [..]" the name here.

File: main/src/main/scala/sbt/Keys.scala [hunk 2]
```diff
@@ -271,199 +271,198 @@ object Keys {
...
+  val transitiveUpdate = taskKey[Seq[UpdateReport]]("UpdateReports for the internal dependencies of this project.").withRank(DTask)
+  val updateClassifiers = TaskKey[UpdateReport]("updateClassifiers", "Resolves and optionally retrieves classified artifacts, such as javadocs and sources, for dependency definitions, transitively.", BPlusTask, update)
+  val transitiveClassifiers = settingKey[Seq[String]]("List of classifiers used for transitively obtaining extra artifacts for sbt or declared dependencies.").withRank(BSetting)
+  val updateSbtClassifiers = TaskKey[UpdateReport]("updateSbtClassifiers", "Resolves and optionally retrieves classifiers, such as javadocs and sources, for sbt, transitively.", BPlusTask, updateClassifiers)
   val sourceArtifactTypes = settingKey[Seq[String]]("Ivy artifact types that correspond to source artifacts. Used by IDEs to resolve these resources.") // BSetting
   val docArtifactTypes = settingKey[Seq[String]]("Ivy artifact types that correspond to javadoc artifacts. Used by IDEs to resolve these resources.") // BSetting
```
Comment (dwijnand) [inline main/src/main/scala/sbt/Keys.scala]: Now that we have it, should we set these ranks?

File: main/src/main/scala/sbt/Keys.scala [hunk 3]
```diff
@@ -271,199 +271,198 @@ object Keys {
...
+  val streams = taskKey[TaskStreams]("Provides streams for logging and persisting data.").withRank(DTask)
   val taskDefinitionKey = Def.taskDefinitionKey
-  val (executionRoots, dummyRoots) = Def.dummy[Seq[ScopedKey[_]]]("execution-roots", "The list of root tasks for this task execution.  Roots are the top-level tasks that were directly requested to be run.")
-
+  val (executionRoots, dummyRoots) = Def.dummy[Seq[ScopedKey[_]]]("executionRoots", "The list of root tasks for this task execution.  Roots are the top-level tasks that were directly requested to be run.")
   val state = Def.stateKey
   val streamsManager = Def.streamsManagerKey
 
-  val stateStreams = AttributeKey[Streams]("streams-manager", "Streams manager, which provides streams for different contexts.  Setting this on State will override the default Streams implementation.")
+  val stateStreams = AttributeKey[Streams]("Streams manager, which provides streams for different contexts.  Setting this on State will override the default Streams implementation.")
```
Comment (dwijnand) [inline main/src/main/scala/sbt/Keys.scala]: Also here, you're setting the description as the name.

**Review by dwijnand — DISMISSED** [review]

File: main/src/main/scala/sbt/Keys.scala (hunk 1 above)
Comment (eed3si9n) [reply main/src/main/scala/sbt/Keys.scala]: Nice catch.

**Comment by eed3si9n** [comment]
All green.

**Review by dwijnand — APPROVED** [review]

**Pull request merged**